indexer/feature_data.cpp

```diff
@@ -220,11 +216,46 @@ uint8_t CalculateHeader(size_t const typesCount, HeaderGeomType const headerGeom
   return header;
 }
 
-void TypesHolder::SortBySpec()
+void TypesHolder::SortByUseless()
 {
   UselessTypesChecker::Instance().SortUselessToEnd(*this);
 }
 
+void TypesHolder::SortBySpec()
+{
+  auto const & cl = classif();
+  int constexpr kZoom = scales::GetUpperStyleScale();
+
+  drule::TypeT arrPointsRules[] = { drule::symbol, drule::caption };
+  drule::TypeT arrLineRules[] = { drule::pathtext };
+  auto rules = arrPointsRules;
+  size_t count = std::size(arrPointsRules);
+  if (m_geomType == GeomType::Line)
+  {
+    rules = arrLineRules;
+    count = std::size(arrLineRules);
+  }
+
+  auto const getPriority = [&](uint32_t type)
+  {
+    drule::KeysT keys;
+    cl.GetObject(type)->GetSuitable(kZoom, m_geomType, keys);
```

Comment:
I think you can just use `cl.GetObject(t)->GetMaxOverlaysPriority();`
No need to check for points vs lines, no need to fix the zoom level (we might use different priorities for different zooms in future).

But wait, why your code is limited to POIs and lines only? What about sorting areas?
(GetMaxOverlaysPriority() doesn't work for areas)

Reply:
Why limited? Point and Area use symbol/caption compare. Line uses pathtext.

Reply:
Or you mean checking area filling priority? Actually, it was not in scope.

Reply:
Is this code is actually used for all PP instances or only for selected POIs and not for e.g. long taps?
Does this code ever mix types from several overlapping features?

Reply:
Is it another problem? Here we select the best type within _one_ Feature. Do not mix it with selecting the best Feature under the point.

Reply:
It doesn't mean we shouldn't fix long tapped selections, right? :)

A single feature can contain several area types, thats why we have this sort in the Stylist.

Reply:
If you don't want to address areas in this PR then makes total sense to use `cl.GetObject(t)->GetMaxOverlaysPriority();` here IMO.
Maybe it'd be better to get back to the areas when fixing
- #4998

Reply:
Ok, then could you clarify some controversial area types examples?
If the area is a pier and building, It is totally fine to show the more specific "pier" type.
I don't have an idea what can be wrong with the current symbol/caption selector.

Reply:
Never mind this case. I was thinking about a big pier with a few buildings on it, which is actually not applicable here (as we're talking about a single feature with multiple types here)..

Reply:
So, ok here to keep only the symbol/caption or not?

Reply:
Yeap, I think it OK for now.
But better to use `cl.GetObject(t)->GetMaxOverlaysPriority();` instead of custom logic.